cemetery/models/cemetery_beneficiary.py

```diff
@@ -0,0 +1,208 @@
+        "cemetery.location", string="Current Location"
+    )
+    complete_name = fields.Char()
+
+    @api.constrains("cemetery_location_id")
+    def _check_cemetery_location_id(self):
+        stock_quant_obj = self.env["stock.quant"]
+        occupied_space = 0
+        cemetery_product = (
+            self.env.company.cemetery_product_template
+            and self.env.company.cemetery_product_template.product_variant_id
+            or self.env["product.product"]
+        )
+        for rec in self:
+            location = rec.cemetery_location_id.location_cemetery_id
+            occupied_space += sum(
+                stock_quant_obj.search([("location_id", "=", location.id)]).mapped(
+                    "quantity"
+                )
+            )
+            storage_cap = location.storage_category_id.mapped(
+                "product_capacity_ids"
+            ).filtered(lambda l: l.product_id.id == cemetery_product.id)
+            available_space = storage_cap.quantity
+            if occupied_space >= available_space:
+                raise ValidationError(
+                    _(
+                        f"Please choose a different location, as this {rec.cemetery_location_id.display_name} one is fully occupied."
+                    )
+                )
+
+    def _prepare_stock_move_vals(self, beneficiary):
+        StockMove = self.env["stock.move"]
+        StockQuant = self.env["stock.quant"]
+        stock_move_list = []
+        cemetery_product = (
+            self.env.company.cemetery_product_template
+            and self.env.company.cemetery_product_template.product_variant_id
+            or self.env["product.product"]
+        )
+        inventory_loss = self.env["stock.location"].search(
+            [
+                ("usage", "=", "inventory"),
+                ("company_id", "=", self.env.company.id),
+                ("active", "=", True),
+            ],
+            limit=1,
+        )
+        location_id = inventory_loss.id
+        location_dest_id = beneficiary.cemetery_location_id.location_cemetery_id.id
+        if self._context.get("source_location") and self._context.get(
+            "desti_location_id"
+        ):
+            location_id = self._context.get("source_location")
+            location_dest_id = self._context.get("desti_location_id")
+        beneficiary.complete_name = (
+            self.env["stock.location"].browse(location_dest_id).complete_name
+        )
+        stock_move_list.append(
+            {
+                "company_id": self.env.company.id,
+                "product_id": cemetery_product.id,
+                "product_uom": cemetery_product.uom_id.id,
+                "location_id": location_id,
+                "location_dest_id": location_dest_id,
+                "name": beneficiary.beneficiary_type,
+                "procure_method": "make_to_stock",
+                "product_uom_qty": 1,
+                "date": beneficiary.death_date,
```

Comment:
In case we have beneficiary_type = rights_holder the death_date is not defined.
How to manage that ?